Library/ViewModels/PledgeViewModelTests.swift

```diff
@@ -3523,4 +3573,83 @@ final class PledgeViewModelTests: TestCase {
 
     XCTAssertEqual(["Pledge Screen Viewed", "Pledge Button Clicked"], self.trackingClient.events)
   }
+
+  func testTrackingEvent_CheckoutProperties_CreditCard() {
+    let createBacking = CreateBackingEnvelope.CreateBacking(
+      checkout: Checkout(state: .verifying, backing: .init(clientSecret: nil, requiresAction: false))
+    )
+    let mockService = MockService(
+      createBackingResult:
+      Result.success(CreateBackingEnvelope(createBacking: createBacking))
+    )
+
+    withEnvironment(apiService: mockService, currentUser: .template) {
+      self.vm.inputs.configureWith(
+        project: .template,
+        reward: Reward.noReward,
+        refTag: .activity,
+        context: .pledge
+      )
+      self.vm.inputs.viewDidLoad()
+
+      self.vm.inputs.creditCardSelected(with: "123")
+
+      self.vm.inputs.pledgeAmountViewControllerDidUpdate(
+        with: (amount: 25.0, min: 10.0, max: 10_000.0, isValid: true)
+      )
+      self.vm.inputs.submitButtonTapped()
+
+      self.scheduler.run()
+
+      XCTAssertEqual(
+        ["Pledge Screen Viewed", "Pledge Button Clicked"],
+        self.trackingClient.events
+      )
+    }
+  }
+
+  func testTrackingEvent_CheckoutProperties_ApplePay() {
+    let createBacking = CreateBackingEnvelope.CreateBacking(
+      checkout: Checkout(state: .successful, backing: .init(clientSecret: nil, requiresAction: false))
+    )
+    let mockService = MockService(
+      createBackingResult:
+      Result.success(CreateBackingEnvelope(createBacking: createBacking))
+    )
+
+    withEnvironment(apiService: mockService, currentUser: .template) {
+      let project = Project.template
+        |> Project.lens.stats.staticUsdRate .~ 2.0
+
+      let reward = Reward.template
+        |> Reward.lens.title .~ "Reward"
+        |> Reward.lens.minimum .~ 5
+        |> Reward.lens.shipping.enabled .~ true
+
+      let shippingRule = ShippingRule.template
+        |> ShippingRule.lens.cost .~ 5
+
+      self.vm.inputs.configureWith(project: project, reward: reward, refTag: .projectPage, context: .pledge)
+      self.vm.inputs.viewDidLoad()
+
+      self.vm.inputs.shippingRuleSelected(shippingRule)
+
+      self.vm.inputs.applePayButtonTapped()
+
+      self.vm.inputs.paymentAuthorizationDidAuthorizePayment(
+        paymentData: (displayName: "Visa 123", network: "Visa", transactionIdentifier: "12345")
+      )
+
+      XCTAssertEqual(
+        PKPaymentAuthorizationStatus.success,
+        self.vm.inputs.stripeTokenCreated(token: "stripe_token", error: nil)
+      )
+
+      self.vm.inputs.paymentAuthorizationViewControllerDidFinish()
+
+      self.scheduler.run()
+
+      XCTAssertEqual(["Pledge Screen Viewed"], self.trackingClient.events)
```

Comment:
Here too... which properties are involved in this test? 🤔

Reply:
Ah right, we actually don't need this test anymore because we don't have the `Completed Checkout` event anymore. Will remove!